src/main/java/io/neonbee/health/HealthCheckRegistry.java

```diff
@@ -173,17 +173,27 @@ Future<List<JsonObject>> getLocalHealthCheckResults() {
         List<Future<JsonObject>> asyncCheckResults =
                 getHealthChecks().values().stream().map(hc -> hc.result().map(CheckResult::toJson)).collect(toList());
 
-        return AsyncHelper.allComposite(asyncCheckResults).map(resolvedCheckResults -> resolvedCheckResults.list()
-                .stream().map(JsonObject.class::cast).peek(result -> result.remove("outcome")).collect(toList()));
+        return AsyncHelper.joinComposite(asyncCheckResults).transform(v -> {
+            List<JsonObject> allSucceeded = asyncCheckResults.stream().filter(Future::succeeded).map(Future::result)
+                    .map(JsonObject.class::cast).peek(result -> result.remove("outcome")).collect(toList());
```

Comment:
I don't see why we need to convert the result to a JsonObject.
```suggestion
                    .peek(result -> result.remove("outcome")).collect(toList());
``